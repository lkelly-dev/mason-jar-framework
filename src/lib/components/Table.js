import React, { useContext } from 'react';
import { cx } from 'emotion';
import { debugAttrs, debugMode } from '../utils/debugging';
// import { Button } from './Button';
// import MODIFIERS from './Modifiers';

const Table = props => {
  const { context, children, className, ...rest } = props;
  const debug = debugMode(context)
  const tableClass = cx('table', className);
  return (
    <table className={tableClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </table>
  );
};

// const NavbarNav = ({ children }) => <nav className="navbar-nav">{children}</nav>;

// const NavbarBrand = props => {
//   const { children } = props;
//   return (
//     <a className="navbar-brand" {...props}>
//       {children}
//     </a>
//   );
// };

// const NavbarCollapse = props => {
//   const { children, className = ' ', ...rest } = props;
//   const NavbarCollapseClass = `collapse navbar-collapse ${className}`;
//   return (
//     <div className={NavbarCollapseClass} {...rest}>
//       {children}
//     </div>
//   );
// };

// const NavbarToggle = props => (
//   <Button className="navbar-toggler" {...props}>
//     <span className="navbar-toggler-icon" />
//   </Button>
// );

// const NavbarText = ({ children }) => <span className="navbar-text">{children}</span>;

// Navbar.Nav = NavbarNav;
// Navbar.Brand = NavbarBrand;
// Navbar.Text = NavbarText;
// Navbar.Collapse = NavbarCollapse;
// Navbar.Toggle = NavbarToggle;

export { Table };
