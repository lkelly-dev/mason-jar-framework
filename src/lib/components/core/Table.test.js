import React from 'react';
import { shallow } from 'enzyme';
import { Table } from "./Table";

describe('Table', () => {
  it('Table matches snapshot', () => {
    const component = shallow(<Table></Table>);
    expect(component).toMatchSnapshot();
  })
});
