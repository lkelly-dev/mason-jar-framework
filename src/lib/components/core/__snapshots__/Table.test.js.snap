// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Table matches snapshot 1`] = `
<table
  className="table"
/>
`;

exports[`Table matches snapshot 1`] = `
<table
  className="table"
/>
`;
